R/PipeOpTaskSurvClassifIPCW.R

```diff
@@ -0,0 +1,137 @@
+#' @title PipeOpTaskSurvClassifIPCW
+#' @name mlr_pipeops_trafotask_survclassif_IPCW
+#' @template param_pipelines
+#'
+#' @description
+#' Transform [TaskSurv] to [TaskClassif][mlr3::TaskClassif] using IPCW (Vock et al., 2016).
+#'
+#' @section Dictionary:
+#' This [PipeOp][mlr3pipelines::PipeOp] can be instantiated via the
+#' [dictionary][mlr3misc::Dictionary] [mlr3pipelines::mlr_pipeops]
+#' or with the associated sugar function [mlr3pipelines::po()]:
+#' ```
+#' PipeOpTaskSurvClassifIPCW$new()
+#' mlr_pipeops$get("trafotask_survclassif_IPCW")
+#' po("trafotask_survclassif_IPCW")
+#' ```
+#'
+#' @section Input and Output Channels:
+#' [PipeOpTaskSurvClassifIPCW] has one input channel named "input", and two
+#' output channels, one named "output" and the other "data".
+#'
+#' Training transforms the "input" [TaskSurv] to a [TaskClassif][mlr3::TaskClassif],
+#' which is the "output".
+#' The target column is named `"status"` and indicates whether an event occurred
+#' in each time interval.
+#' The transformed task now has the property "weights".
+#' The "data" is NULL.
+#'
+#' During prediction, the "input" [TaskSurv] is transformed to the "output"
+#' [TaskClassif][mlr3::TaskClassif] with `"status"` as target.
+#' The "data" is a [data.table] containing the "time" of each subject as well
+#' as corresponding "row_ids".
+#' This "data" is only meant to be used with the [PipeOpPredClassifSurvIPCW].
+#'
+#' @section Parameters:
+#' The parameters are
+#'
+#' * `cutoff_time :: numeric()`\cr
+#' Cutoff time for IPCW. Observations with time larger than `cutoff_time` are censored.
+#' Should be reasonably smaller than the maximum event time to avoid enormous weights.
+#' * `eps :: numeric()`\cr
+#' Small value to replace `0` survival probabilities with to prevent infinite weights.
+#'
+#' @references
+#' `r format_bib("vock_2016")`
+#'
+#' @family PipeOps
+#' @family Transformation PipeOps
+#' @export
+PipeOpTaskSurvClassifIPCW = R6Class(
+  "PipeOpTaskSurvClassifIPCW",
+  inherit = mlr3pipelines::PipeOp,
+
+  public = list(
+    #' @description
+    #' Creates a new instance of this [R6][R6::R6Class] class.
+    initialize = function(id = "trafotask_survclassif_IPCW") {
+      param_set = ps(
+        cutoff_time = p_dbl(lower = 0, special_vals = list()),
+        eps = p_dbl(lower = 0, default = 1e-6)
```

Comment:
too low `eps`

Reply:
What do you recommend?

Reply:
`eps` here is the G(t), censoring surv probability, a value of 0.001 corresponding to a weight of 1000 seems a better choice to me.